subsys/mpsl/cx/mpsl_cx.c

```diff
@@ -10,12 +10,19 @@
 #if IS_ENABLED(CONFIG_MPSL_CX_THREAD)
 #include <mpsl/mpsl_cx_config_thread.h>
 
+#if DT_HAS_COMPAT_STATUS_OKAY(generic_radio_coex_three_wire)
+#define CX_NODE DT_COMPAT_GET_ANY_STATUS_OKAY(generic_radio_coex_three_wire)
+#else
+#define CX_NODE DT_INVALID_NODE
+#error No "generic-radio-coex-three-wire"-compatible (and enabled) nodes found
+#endif
+
 static int cx_thread_configure(void)
 {
 	struct mpsl_cx_thread_interface_config cfg = {
-		.request_pin  = CONFIG_MPSL_CX_THREAD_PIN_REQUEST,
-		.priority_pin = CONFIG_MPSL_CX_THREAD_PIN_PRIORITY,
-		.granted_pin  = CONFIG_MPSL_CX_THREAD_PIN_GRANT,
+		.request_pin  = DT_GPIO_PIN(CX_NODE, req_gpios),
+		.priority_pin = DT_GPIO_PIN(CX_NODE, pri_dir_gpios),
+		.granted_pin  = DT_GPIO_PIN(CX_NODE, grant_gpios),
```

Comment:
This is changing a PSEL value (port << 5 | pin) to a pin, which means it will not work for P1.x pins.

Take a look at these options instead:

https://github.com/nrfconnect/sdk-zephyr/blob/379117545b61f57d9263fc3f0048929c779524df/soc/arm/nordic_nrf/soc_nrf_common.h#L94
https://github.com/nrfconnect/sdk-zephyr/blob/379117545b61f57d9263fc3f0048929c779524df/include/drivers/gpio.h#L327

Reply:
nice catch, I definitely wouldn't have thought to check this. I'll keep it in mind for future changes